extensions/ql-vscode/src/common/zlib.ts

```diff
@@ -0,0 +1,43 @@
+import { gzip, gunzip, InputType as ZlibInputType, ZlibOptions } from "zlib";
+
+/**
+ * Promisified version of zlib.gzip
+ * @param buffer Buffer to compress
+ * @param options zlib options
+ */
+export function gzipEncode(
```

Comment:
Can we use [promisify](https://nodejs.org/api/util.html#utilpromisifyoriginal) instead?